packages/flutter/lib/src/rendering/paragraph.dart

```diff
@@ -282,9 +436,88 @@ class RenderParagraph extends RenderBox {
   @visibleForTesting
   bool get debugHasOverflowShader => _overflowShader != null;
 
+  void _layoutText({ double minWidth = 0.0, double maxWidth = double.infinity }) {
+    final bool widthMatters = softWrap || overflow == TextOverflow.ellipsis;
+    _textPainter.layout(minWidth: minWidth, maxWidth: widthMatters ? maxWidth : double.infinity);
+  }
+
+  void _layoutTextWithConstraints(BoxConstraints constraints) {
+    _layoutText(minWidth: constraints.minWidth, maxWidth: constraints.maxWidth);
+  }
+
+  // Layout the child inline widgets. We then pass the dimensions of the
+  // children to _textPainter so that appropriate placeholders can be inserted
+  // into the LibTxt layout. This does not do anything if no inline widgets were
+  // specified.
+  void _layoutChildren(BoxConstraints constraints) {
+    RenderBox child = firstChild;
+    final List<PlaceholderDimensions> placeholderDimensions = List<PlaceholderDimensions>(childCount);
+    int childIndex = 0;
+    while (child != null) {
+      // Set min constraints to 0, since the text min constraints don't apply
+      // to the inline widgets.
+      child.layout(
+        BoxConstraints(
+          minWidth: 0,
+          minHeight: 0,
+          maxWidth: constraints.maxWidth,
+          maxHeight: constraints.maxHeight
+        ),
```

Comment:
you can just pass `constraints.loose` or some such to get this effect.

That said, are you sure these are the right constraints? Consider a paragraph with two placeholders, both SizedBox.expand.